### PR TITLE
Accessblocker protects write components by checking auth and vpn

### PR DESCRIPTION
**Pull request by peterbe**
It started with https://github.com/mozilla/delivery-console/pull/795
Each component, whose existence in life is for preparing for making a write-XHR (aka. POST) had to get information about whether the user is logged in and whether the VPN check passed. And then, in each of those components, it has check these things in the render function and potentially "redirect" the render to `<AuthenticationAlert/>` or `<VPNAlert/>`. Tedious. So instead, wrap each of these components in a higher-order component that checks this once.

**No tests added yet.**

### CODE REVIEW

**Comment by peterbe** [comment]
CC @mythmon

**Comment by peterbe** [comment]
Another *potential* use we can have with this is to make it pass along the `userProfile` and `vpnAvailable`. Then, components like `DetailsBar` would get those in as props and `DetailsBar` would be able to do something like this:

```javascript
class DetailsBar {
  render() {
    const { userProfile, vpnAvailable } = this.props
    return <div>
      <Button disabled={!userProfile || !vpnAvailable}>Edit</Button>
    </div>
  }
}

export default AccessBlocker(DetailsBar);
```

Same goes for the ApprovalRequest component. This would effecticly replace (and extend with the VPN check) this code: https://github.com/mozilla/delivery-console/blob/ed084e4b242cd1d531b19fbbec33742088ba5bed/src/workflows/recipes/components/ApprovalRequest.js#L112-L128

File: src/components/common/AccessBlocker.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+function AccessBlocker(ProtectedComponent) {
+  @connect(
+    state => {
+      return { userProfile: getUserProfile(state), vpnAvailable: isNormandyAdminAvailable(state) };
+    },
+    {},
+  )
+  class AccessBlockerInner extends React.Component {
+    static propTypes = {
+      // XXX How do you make this `null | instanceOf(Map)`??
```
Comment (rehandalal) [inline src/components/common/AccessBlocker.js]: you could do something along the lines of:
```js
PropTypes.oneOf([
  PropTypes.instanceOf(Map),
  PropTypes.instanceOf(null),
]),
```

although i always thought if it didn't have `.isRequired` then the `null` should be fine.

File: src/components/common/AccessBlocker.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      //   userProfile: PropTypes.instanceOf(Map),
+      vpnAvailable: PropTypes.bool,
+    };
+
+    render() {
+      const { userProfile, vpnAvailable, ...passThroughProps } = this.props;
+      // console.log(JSON.stringify(userProfile));
```
Comment (rehandalal) [inline src/components/common/AccessBlocker.js]: nit: cleanup

File: src/components/common/AccessBlocker.js (hunk 1 above)
Comment (peterbe) [reply src/components/common/AccessBlocker.js]: Maybe there's a difference between absent and `null` and PropTypes does something like:
```javascript
if (value !== undefined) {
  check(value, expectedType)
}
```

More importantly, I fear there's some lack of understanding here. 

If you put this same `@connect(state => { return {userProfile: getUserProfile(state)} ...)` on the `CreateRecipePage` component instead, then it's never null. ...if you're logged in. 

But when it's here on the `AccessBlockerInner` it's `null` briefly and whence the auth-js check is complete it becomes an actual `Map` object.

Comment (peterbe) [reply src/components/common/AccessBlocker.js]: @rehandalal Would you mind taking a look and try to find out why, when logged in, the `userProfile` is never `null` when used on page components like `CreateRecipePage` but *can be* on the `AccessBlocker` wrapper component.

**Comment by rehandalal** [comment]
Closing for now

**Pull request closed without merging**